### PR TITLE
BUG: make time type string [s ns]

### LINKED ISSUE #642: Invalid default value for elements of type <time>
In `types.xsd` time is defined as a simple double, however inside `state.sdf` a string value of the form `"0 0"` is used as default. The documentation agrees with the latter and says it should be of the form `"{seconds} {nanoseconds}`.

I implemented a local fix inside `types.xsd` of the form

```xml
  <xsd:simpleType name="time">
    <xsd:restriction base="xsd:string">
      <xsd:pattern value="\d+ \d+"/>
    </xsd:restriction>
  </xsd:simpleType>
```

However, it still makes sense to bring it up as an issue here, since I don't know if this is the preferred solution.

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
# 🦟 Bug fix

Fixes #642 

## Summary
While I did this in #643 I think it makes sense to make it a separate PR and rebase. This way this bug can be fixed while awaiting feedback for different things in the other PR.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by chapulina — APPROVED** [review]

**Comment by FirefoxMetzger** [comment]
This is such an odd failure. `double_pendulum.sdf` doesn't even use a state element and by extension neither a time element, which is the only one I touched.

The log seems to agree with me:
```
```

Here is the relevant section from the jenkins log:

```
test 103
        Start 103: INTEGRATION_schema_test

103: Test command: /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/build/test/integration/INTEGRATION_schema_test "--gtest_output=xml:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/build/test_results/INTEGRATION_schema_test.xml"
103: Environment variables: 
103:  SDF_PATH=/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.8:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.7:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.6:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.5:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.4:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.3:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.2:/Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/sdf/1.0:
103: Test timeout computed to be: 240
103: Running main() from gtest_main.cc
103: [==========] Running 3 tests from 1 test case.
103: [----------] Global test environment set-up.
103: [----------] 3 tests from SDFSchemaGenerator
103: [ RUN      ] SDFSchemaGenerator.TestDoublePendulum
103: error : Operation in progress
103: warning: failed to load external entity "http://sdformat.org/schemas/rfidtag.xsd"
103: http://sdformat.org/schemas/sensor.xsd:19: element include: Schemas parser error : Element '{http://www.w3.org/2001/XMLSchema}include': Failed to load the document 'http://sdformat.org/schemas/rfidtag.xsd' for inclusion.
103: WXS schema /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/build/sdf/1.8/root.xsd failed to compile
103: CMD[xmllint --noout --schema /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/build/sdf/1.8/root.xsd /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/test/integration/model/double_pendulum.sdf]
103: /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/test/integration/schema_test.cc:52: Failure
103: Failed
103: Fail in parsing the model
103: [  FAILED  ] SDFSchemaGenerator.TestDoublePendulum (62365 ms)
103: [ RUN      ] SDFSchemaGenerator.TestPR2Model
103: /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/test/integration/model/pr2.sdf validates
103: CMD[xmllint --noout --schema /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/build/sdf/1.8/root.xsd /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/test/integration/model/pr2.sdf]
103: [       OK ] SDFSchemaGenerator.TestPR2Model (3036 ms)
103: [ RUN      ] SDFSchemaGenerator.TestTurtleBotModel
103: /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/test/integration/model/turtlebot.sdf validates
103: CMD[xmllint --noout --schema /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/build/sdf/1.8/root.xsd /Users/jenkins/workspace/sdformat-ci-pr_any-homebrew-amd64/sdformat/test/integration/model/turtlebot.sdf]
103: [       OK ] SDFSchemaGenerator.TestTurtleBotModel (2949 ms)
103: [----------] 3 tests from SDFSchemaGenerator (68350 ms total)
103: 
103: [----------] Global test environment tear-down
103: [==========] 3 tests from 1 test case ran. (68350 ms total)
103: [  PASSED  ] 2 tests.
103: [  FAILED  ] 1 test, listed below:
103: [  FAILED  ] SDFSchemaGenerator.TestDoublePendulum
103: 
103:  1 FAILED TEST
103/216 Test #103: INTEGRATION_schema_test ................................***Failed   68.36 sec
```

**Comment by chapulina** [comment]
humm strange failure, only on macOS. There aren't known flakes for this branch and platform, that's why we set the check as `Required`.

Issues on macOS could be due to different dependency versions... Sometimes it's a clang vs gcc thing, or a timing issue (because our macOS CI machines are slow), but I don't think it could be those here.

**Comment by FirefoxMetzger** [comment]
Okay. It could indeed be a timeout, considering that the test took ~ 1 minute to fail for such a simple thing (running xmllint should take ms, right?). I'm not a Jenkins user myself, but is the setup included in this time? If not, this could be further evidence for a timeout.

File: sdf/1.5/schema/types.xsd [hunk 1]
```diff
@@ -31,7 +31,8 @@
   </xsd:simpleType>
 
   <xsd:simpleType name="time">
-    <xsd:restriction base="xsd:double">
+    <xsd:restriction base="xsd:string">
+      <xsd:pattern value="\d+ \d+"/>
```
Comment (jennuine) [inline sdf/1.5/schema/types.xsd:35]: Could you also update the others as well

```suggestion
      <xsd:pattern value="\s*\d+\s+\d+\s*"/>
```

**Review by jennuine — CHANGES_REQUESTED** [review]
Thanks for the fix! Just a quick comment

File: sdf/1.5/schema/types.xsd (hunk 1 above)
Comment (FirefoxMetzger) [reply sdf/1.5/schema/types.xsd:35]: @jennuine Hm, if we are worried about whitespace, wouldn't something like this be a bit more readable?

```
<xsd:restriction base="xsd:string">
  <xsd:whiteSpace value="collapse"/>
  <xsd:pattern value="\d+ \d+"/>
</xsd:restriction>
```

Comment (jennuine) [reply sdf/1.5/schema/types.xsd:35]: Sounds good to me :+1:

Comment (FirefoxMetzger) [reply sdf/1.5/schema/types.xsd:36]: @jennuine Done.

**Review by jennuine — APPROVED** [review]

**Pull request merged**